Tests/PointFreeTests/GhostTests.swift

```diff
@@ -0,0 +1,191 @@
+import Either
+import HttpPipeline
+import Models
+import Optics
+@testable import PointFree
+import PointFreePrelude
+import PointFreeRouter
+import PointFreeTestSupport
+import Prelude
+import SnapshotTesting
+import XCTest
+
+final class GhostTests: TestCase {
+  override func setUp() {
+    super.setUp()
+    record=true
+  }
+
+  func testStartGhosting_HappyPath() {
+    let adminUser = User.admin
+    let adminSession = Session.loggedIn
+      |> \.user .~ .standard(adminUser.id)
+
+    let ghostee = User.mock
+      |> \.id .~ User.Id(rawValue: UUID(uuidString: "10101010-dead-beef-dead-beefdeadbeef")!)
+
+    update(
+      &Current,
+      (\Environment.database) .~ .mock,
+      (\Environment.database.fetchUserById) .~ { userId -> EitherIO<Error, User?> in
+        pure(
+          userId == adminUser.id ? adminUser
+            : userId == ghostee.id ? ghostee
+            : nil
+        )
+      }
+    )
+
+    let conn = connection(
+      from: request(to: .admin(.ghost(.start(ghostee.id))), session: adminSession)
+      )
+      |> siteMiddleware
+      |> Prelude.perform
+
+    _assertInlineSnapshot(matching: conn, as: .conn, with: """
+POST http://localhost:8080/admin/ghost/start
+Cookie: pf_session={"userId":"12121212-1212-1212-1212-121212121212"}
+
+user_id=10101010-DEAD-BEEF-DEAD-BEEFDEADBEEF
+
+302 Found
+Location: /
+Referrer-Policy: strict-origin-when-cross-origin
+Set-Cookie: pf_session={"user":{"ghosteeId":"10101010-DEAD-BEEF-DEAD-BEEFDEADBEEF","ghosterId":"12121212-1212-1212-1212-121212121212"}}; Expires=Sat, 29 Jan 2028 00:00:00 GMT; Path=/
```

Comment:
this test shows our session transitioning into a ghosting session when we hit the `/admin/ghost/start` endpoint.